### ISSUE TITLE
Cannot mock destructors in Visual Studio 2015 C++

### ISSUE THREAD

**Issue by francoisgenolini** [eranpeer/FakeIt#54]
On Windows 10 (latest update), I have a Visual Studio 2015 native C++ unit test project and I am mocking destructors using FakeIt.
This results in access violations:
fakeit.hpp, line 5806. fakeit::MethodProxyCreator<void>::methodProxy(unsigned int id)
where `invocationHandlerCollection` is an invalid object

Here is my code:

``` C++
        struct SomeInterface {
            virtual ~SomeInterface() {}
            virtual int SomeMethod(int someArgument) = 0;
        };

            Mock<SomeInterface> mock;
            When(Method(mock, SomeMethod)).AlwaysReturn(42);
            Fake(Dtor(mock));
            SomeInterface* someInterface = &mock.get();
            // This next line causes access violation
            delete someInterface;
```

**Comment by st4lk3r87** [comment]
Hello, did you find any solution to this? I've a similar issue on visual studio 2013 but the crash happens on 
```c++
        void *VirtualTableBase::getCookie(int index) {
            return _firstMethod[-2 - index];
        }
```

**Comment by tnovotny** [comment]
For me this is working. I am using [the November Update 1](https://www.visualstudio.com/en-us/news/vs2015-update1-vs.aspx) and the MS testing tool. Test passes both x32 and x64.

**Comment by alexey-malov** [comment]
I have checked with Visual Studio 2015 update 3.
It crashes if destructor is not the very first method of the interfaces:

If I place SomeMethod after the destructor, everything is OK:
```c++
	struct SomeInterface
	{
		virtual ~SomeInterface() = default;
		virtual void SomeMethod(int someParam) = 0;
	};

	void mock_virtual_dtor_with_when() {
		int a = 0;
		Mock<SomeInterface> mock;
		When(Dtor(mock)).Return().Do([&](){a++; });
		SomeInterface * i = &(mock.get());
		delete i; // do nothing
		delete i; // a++
		ASSERT_EQUAL(1, a);
	}

```

However, if I put SomeMethod above the destructor, it crashes:

```c++
	struct SomeInterface
	{
		virtual void SomeMethod(int someParam) = 0;
		virtual ~SomeInterface() = default;
	};
	void mock_virtual_dtor_with_when() {
		int a = 0;
		Mock<SomeInterface> mock;
		When(Dtor(mock)).Return().Do([&](){a++; });
		SomeInterface * i = &(mock.get());
		delete i; // do nothing
		delete i; // a++
		ASSERT_EQUAL(1, a);
	}
```

**Comment by FranckRJ** [comment]
Should be fixed by #312.
